File: Python3/3264. Final Array State After K Multiplication Operations I.py
class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        heap = [(n, i) for i, n in enumerate(nums)]
        heapify(heap)

        for _ in range(k):
            n, i = heappop(heap)
            nums[i] = n * multiplier
            heappush(heap, (nums[i], i))

        return nums

File: Java/3264. Final Array State After K Multiplication Operations I.java
class Solution {
    public int[] getFinalState(int[] nums, int k, int multiplier) {
        PriorityQueue<int[]> heap = new PriorityQueue<>((a, b) -> {
            int value = Integer.compare(a[0], b[0]);
            if (value == 0) {
                return Integer.compare(a[1], b[1]);
            }
            return value;
        });

        for (int i = 0; i < nums.length; i++) {
            heap.offer(new int[] {nums[i], i});
        }

        while (k-- > 0) {
            int[] cur = heap.poll();
            int n = cur[0], i = cur[1];
            nums[i] *= multiplier;
            heap.offer(new int[] {nums[i], i});
        }

        return nums;
    }
}